app/src/main/java/com/ideastorm/v25001/MainViewModel.kt

```
@@ -9,12 +9,17 @@ import com.ideastorm.v25001.service.IActivityService
 import kotlinx.coroutines.launch
 
 class MainViewModel(var activityService : IActivityService = ActivityService()) : ViewModel() {
-    var activity : MutableLiveData<Activity> = MutableLiveData<Activity>()
+    private var activity : MutableLiveData<Activity> = MutableLiveData<Activity>()
 
     fun fetchActivity() {
-        viewModelScope.launch {
-            var innerActivity = activityService.fetchActivity()
-            activity.postValue(innerActivity)
+        try {
+            viewModelScope.launch {
+                var innerActivity = activityService.fetchActivity()
+                activity.postValue(innerActivity)
+            }
+        } catch (e: Exception) {
+            throw IllegalArgumentException(e)
```

Catching and re-throwing an exception isn't adding any value.

If adding a catch block, heal the exception there.  At the very least, add a logging statement.  An empty catch block is the worst possible option, as you're losing valuable information about why something crashed, and assuming that everything worked fine.